android/app/src/main/java/com/expensify/chat/CustomNotificationProvider.java

```diff
@@ -0,0 +1,234 @@
+package com.expensify.chat;
+
+import android.content.Context;
+import android.graphics.Bitmap;
+import android.os.Build;
+import android.util.DisplayMetrics;
+import android.util.Log;
+import android.util.TypedValue;
+import android.view.WindowManager;
+
+import androidx.annotation.DrawableRes;
+import androidx.annotation.NonNull;
+import androidx.core.app.NotificationCompat;
+import androidx.core.app.NotificationManagerCompat;
+import androidx.core.app.Person;
+import androidx.core.graphics.drawable.IconCompat;
+
+import com.urbanairship.AirshipConfigOptions;
+import com.urbanairship.json.JsonMap;
+import com.urbanairship.json.JsonValue;
+import com.urbanairship.push.PushMessage;
+import com.urbanairship.push.notifications.NotificationArguments;
+import com.urbanairship.reactnative.ReactNotificationProvider;
+import com.urbanairship.util.ImageUtils;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+
+public class CustomNotificationProvider extends ReactNotificationProvider {
+
+    // Resize icons to 100 dp x 100 dp
+    private static final int MAX_ICON_SIZE_DPS = 100;
+
+    // Max wait time to resolve an icon. We have ~10 seconds to a little less
+    // to ensure the notification builds.
+    private static final int MAX_ICON_FETCH_WAIT_TIME_SECONDS = 8;
+
+    // Fallback drawable ID. 0 to not use a fallback ID.
+    private static final int FALLBACK_ICON_ID = 0;
+
+    // Logging
+    private static final String TAG = "NotificationProvider";
+
+    // Conversation JSON keys
+    private static final String PAYLOAD_KEY = "payload";
+    private static final String TYPE_KEY = "type";
+    private static final String REPORT_COMMENT_TYPE = "reportComment";
+
+    private final ExecutorService executorService = Executors.newCachedThreadPool();
+    public final HashMap<Integer, NotificationCache> cache = new HashMap<>();
+
+    public CustomNotificationProvider(@NonNull Context context, @NonNull AirshipConfigOptions configOptions) {
+        super(context, configOptions);
+    }
+
+    @NonNull
+    @Override
+    protected NotificationCompat.Builder onExtendBuilder(@NonNull Context context, @NonNull NotificationCompat.Builder builder, @NonNull NotificationArguments arguments) {
+        super.onExtendBuilder(context, builder, arguments);
+        PushMessage message = arguments.getMessage();
+
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
+            return builder;
+        }
```

Comment:
I'm guessing we're not supporting Android Lollipop here due to [this Marshmallow Notification Icon feature](https://developer.android.com/about/versions/marshmallow/android-6.0?hl=en#notifications)? Or is there another reason that we are unable to style notifications for devices running lower than M? 

I would have thought that `IconCompat` should be backward compatible.

Reply:
It turns out we don't need this!